src/wp-includes/blocks.php

```diff
@@ -438,26 +444,47 @@ function register_block_type_from_metadata( $file_or_folder, $args = array() ) {
 		'editorStyle' => 'editor_style_handles',
 		'style'       => 'style_handles',
 	);
+	$version      = ! $is_core_block && isset( $metadata['version'] ) ? $metadata['version'] : false;
 	foreach ( $style_fields as $metadata_field_name => $settings_field_name ) {
 		if ( ! empty( $metadata[ $metadata_field_name ] ) ) {
 			$styles           = $metadata[ $metadata_field_name ];
 			$processed_styles = array();
 			if ( is_array( $styles ) ) {
 				for ( $index = 0; $index < count( $styles ); $index++ ) {
-					$result = register_block_style_handle(
-						$metadata,
-						$metadata_field_name,
-						$index
-					);
+					if ( $is_core_block && in_array( $metadata_field_name, $default_styles, true ) ) {
+						$style_handle = generate_block_asset_handle( $metadata['name'], $metadata_field_name, $index );
+						$result       = wp_register_style(
+							$style_handle,
+							false,
+							array(),
+							$version
+						);
+					} else {
```

Comment:
It would be nice to keep all of the logic for registering the block styles in `register_block_style_handle()` rather than duplicating some here. Could this logic be handled inside that function instead?

Reply:
@joemcgill I tried exploring this here https://github.com/spacedmonkey/wordpress-develop/commit/3eb647298a86ae3099059effb316458bc4cb0efc. What you do think?